### PR TITLE
feat(tests): Added tests for the tfe block

### PR DESCRIPTION
**Pull request by ghost**
This PR resolves issue #90.

Whats has changed/added in this PR

- Fixed a bug with the `workspace_env_vars` in the TFE block. When the module was trying to merge the variables for the TFC workspace, it was incorrectly attempting to merge content from an undefined variable `var.tfe.env_vars`. 
- Renamed all tests (in `test` folder) to start with `test_` 
- Created a test for the tfe block in the new offical terraform testing framework. [docs](https://developer.hashicorp.com/terraform/language/tests). This is located inside the `tests` folder. See new readme on how to run the test
- Added an output block in the tfe module to be able to test the creation of workspace variables 
- Changed the role_definitions module to allow not specifying scope. If no scope is provided it will use the current subscription as the default scope. This was done to prevent us for specifying a subscription ID in the tests.

### CODE REVIEW

File: role_definitions.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ resource "azurerm_role_definition" "user_created" {
   role_definition_id = each.value.role_definition_id
   name               = each.value.name
   description        = each.value.description
-  scope              = each.value.scope
+  scope              = each.value.scope == null ? "/subscriptions/${data.azurerm_client_config.current.subscription_id}" : each.value.scope
```
Comment (kimfy) [inline role_definitions.tf]: I want to discuss if we can set the default scope to the workload's resource group id. Are there any downsides to doing that @SanderBlom?

Comment (SanderBlom) [reply role_definitions.tf]: It would probably be a better to reduce the scope to the workload resource group. I'll update the PR :)

**Comment by SanderBlom** [comment]
I reached out to Hashicorp about this issue and this was the response I got back: 

----
I completely understand your limitation, and we are working on enhancing how variables are done in the test framework. It might not happen asap, might be a few months, but we know this is a problem and intend on building the capability. 

Please open a github issue and we will prioritize this. We intend on making many more changes to the test framework, so we will work on this for sure.

Omar Ismail
Senior Product Manager, Terraform Core
[oismail@hashicorp.com](mailto:oismail@hashicorp.com)

**Comment by SanderBlom** [comment]
I have now opened an issue on this https://github.com/hashicorp/terraform/issues/34534

**Review by SanderBlom — APPROVED** [review]

**Pull request merged**